modules/client/src/main/java/org/apache/ignite/internal/client/ClientTransactionInflights.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.client;
+
+import static java.util.concurrent.CompletableFuture.failedFuture;
+import static org.apache.ignite.internal.lang.IgniteStringFormatter.format;
+import static org.apache.ignite.internal.util.CompletableFutures.nullCompletedFuture;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
+
+/**
+ * Client transaction inflights tracker.
+ */
+public class ClientTransactionInflights {
+    /** Hint for maximum concurrent txns. */
+    private static final int MAX_CONCURRENT_TXNS = 1024;
```

Comment:
- Misleading name, this is not a limit, but initial capacity
- 1024 seems too much, I would remove this altogether and use the default map ctor

Reply:
Renamed to MAX_CONCURRENT_TXNS_HINT to better reflect the javadoc.
This is the estimation of maximum number of infligh transaction.
1024 seems ok to me and can easily be achieved on a single client.
We have the same setting for server-side transactions: org.apache.ignite.internal.tx.impl.TransactionInflights#MAX_CONCURRENT_TXNS
Because the tracking map is accessed on a hot path, I want to avoid all map resizes if possible.